### ISSUE TITLE
Site settings shows the wp.com username for a Jetpack site

### ISSUE THREAD

**Issue by koke** [wordpress-mobile/WordPress-iOS#3719]
When using Jetpack REST, the blog's `account` we don't know the self hosted username, so it's showing the wp.com username.

**Comment by koke** [comment]
After #3860, this doesn't show the wp.com username. It shows 'Enter username', which I think it's worse, since the field is not even editable.

![site-settings](https://cloud.githubusercontent.com/assets/8739/8180916/174ade5c-1422-11e5-9dd1-e073defe8fda.png)

This is because when we get a Jetpack site from WordPress.com, we don't know the username that's connected, and I'm not sure if there's a way to find out.

Which made me reconsider the whole screen. I think this screen is mostly legacy stuff that never got reconsidered. Back in the day, you could edit the username/password from there. For reasons I can't remember (but probably related to simplicity), we made only the password editable. Right now we don't need to display the URL (already visible in the list and blog details), or username (only because you can add a site twice with different accounts).
Geotagging could be made an app-wide setting, but then what do we do with users that have it enabled only for some sites?

Then there's the question of how do you change a site's password in the app, other than removing and adding the site again.
Some parts of the app will let you know when the password is wrong, but not how to change it:
![ios simulator screen shot 16 jun 2015 12 31 55](https://cloud.githubusercontent.com/assets/8739/8181136/c4d2ebcc-1423-11e5-8f45-2aaa8354222c.png)

Others, not even that:

![ios simulator screen shot 16 jun 2015 12 33 09](https://cloud.githubusercontent.com/assets/8739/8181154/ecd51ab4-1423-11e5-86ae-f10fb998a870.png)
![ios simulator screen shot 16 jun 2015 12 33 24](https://cloud.githubusercontent.com/assets/8739/8181153/ecd33d34-1423-11e5-9636-bc6aaef50ccc.png)

This probably needs a separate issue(s), but I needed to write it down somewhere

**Comment by folletto** [comment]
> Which made me reconsider the whole screen.

Excellent thinking here. ;)

I'd add that on my screen I see also the "Push Notifications" settings below "Geotagging".

—

Since one of our objectives is creating a uniform experience, I'd also cross-reference in this discussion the settings page at https://wordpress.com/settings/general 

What I would do there is:

1. For "WordPress.com" — Tell that it's getting that from the WordPress.com account. It's ok to show the URL too. 
2. For "Jetpack" — In the same way as the above, it has to say "Jetpack" in some way. And should add that's linked through WordPress.com
3. For "Self Hosted" — Say it, and show all the editable fields as it is now. :)

I'd also add that it would be nice if you could change the Site title, Tagline and Language from there, which are basics things. :)

—

> Some parts of the app will let you know when the password is wrong, but not how to change it:

This is definitely a separate issue. Worth a different ticket. 

Should probably always show the same pop-up leading to a screen where to solve the issue as you pointed out. :)

**Comment by koke** [comment]
> I'd add that on my screen I see also the "Push Notifications"

Correct, that's available for WordPress.com/Jetpack sites, it just doesn't show up in the simulator

**Comment by koke** [comment]
I'm not sure I agree about showing WordPress.com/Jetpack/Self hosted. IMO that's an implementation detail only for us to worry about.

My bigger question was if we really need to show the URL and username there. In my mind those aren't "settings", that's information about the blog and it belongs in the previous screen, if anything.

![ios simulator screen shot 16 jun 2015 13 25 17](https://cloud.githubusercontent.com/assets/8739/8182002/2f808ce8-142b-11e5-9b40-97d721a2737a.png)

Perhaps we stop showing usernames, and don't allow adding a site twice (which would also fix #2782)

**Comment by folletto** [comment]
For me it's relevant because it's a low-level page (My Sites → Site → Settings) and telling in some way where is that account coming from is useful. Plus, it's another point to show Jetpack. ;)

I agree that's not imperative to show usernames, but I feel that "somehow" it needs to say where that account comes from. Being a username, a Jetpack indication or else. :)

**Comment by folletto** [comment]
So, here's an idea:

![mysites-settings-i1](https://cloud.githubusercontent.com/assets/4389/8203993/7b85aa6c-14dd-11e5-9c88-06793f10aacd.png)

I took your feedback on the need of a username there, and removed it, making basically the screens for both .com and Jetpack sites identical with just a change in the small title so you get a "hint" of the source of the site (WordPress.com or Jetpack).

Since Self-Hosted require special attention, I introduced a separate section just for them, so we can manage it in a clean way. :) 

Oh, also: dark text is editable, light is not editable. So in theory here you'd be able to customize the site title and tagline. ;)

How does it look?

**Comment by koke** [comment]
I like how that looks :grinning: Making the title/tagline editable would require some backend changes but it's definitely doable.

But this surfaced another issue: when to save.

Currently, if you toggle any of the switches (geotagging or push notifications), the changes take effect immediately. When you edit the password, a "Save" button appears in the top right corner

![settings-save](https://cloud.githubusercontent.com/assets/8739/8204671/8c34a856-14ea-11e5-91b8-ee9067bd6ffb.png)

If you go back without tapping Save, the changes are lost :disappointed: I don't think we can work around this other than making the settings modal.

The title/tagline fields are less problematic, since we could update them when the field loses focus or when we navigate away from the view

Although going back to consistency with https://wordpress.com/settings/general I wouldn't mind having a "Save" button for all options

**Comment by folletto** [comment]
> Making the title/tagline editable would require some backend changes but it's definitely doable.

We can do it in two steps, so we don't have immediately to add that kind of backend change. ;)

> I wouldn't mind having a "Save" button for all options

Can we just start by using always, all the time, the explicit "Save" (just because seems technically simpler) and then review that later? ;)

**Comment by tonyr59h** [comment]
Addressed in #3897, closing.
